# Plastid genome of Engelmann spruce (*Picea engelmannii*), genotype Se404-851
ðŸŒ² Annotation of the plastid genome of Engelmann spruce (*Picea engelmannii*), genotype Se404-851
[https://www.ncbi.nlm.nih.gov/nuccore/MK241981](https://www.ncbi.nlm.nih.gov/nuccore/MK241981)

![Se404-851 chloroplast genome] (https://github.com/bcgsc/picea-engelmannii-plastid/blob/master/ogdraw/Se404-851cp-outfile-preview.png)

## GeSeq
The assembled FASTA file was inputed into [GeSeq](https://chlorobox.mpimp-golm.mpg.de/geseq.html).
* see [Documentation](https://academic.oup.com/nar/article/45/W1/W6/3806659)
* [GeSeq Settings](https://github.com/bcgsc/picea-glauca-plastid/blob/master/GeSeq/GeSeq_settings.png)
  * The settings used for this annotation are the exact same as those used for *Picea glauca* WS77111 [chloroplast](https://github.com/bcgsc/picea-glauca-plastid)
* GeSeq generates an [OGDRAW .png file](https://github.com/bcgsc/picea-engelmannii-plastid/blob/master/GeSeq/Se404-851cp_OGDRAW.jpg) and a [GenBank .gb file](https://github.com/bcgsc/picea-engelmannii-plastid/blob/master/GeSeq/Se404-851cp.gb).

## EMBOSS Seqret
The .gb file is converted to a [.gff file](https://github.com/bcgsc/picea-engelmannii-plastid/blob/master/Se404-851cp.gff) using [EMBOSS Seqret](https://www.ebi.ac.uk/Tools/sfc/emboss_seqret/):
* [EMBOSS Seqret settings](https://github.com/bcgsc/picea-glauca-plastid/blob/master/EMBOSS-setings.png)
  * The settings used for this conversion are the exact same as those used for *Picea glauca* WS77111 [chloroplast](https://github.com/bcgsc/picea-glauca-plastid)

## Manual Annotation
Duplicate annotations made by GeSeq were removed. Most conflicts were due to the use of multiple reference annotations, as the whole taxus was selected. tRNAs with anti-codons annotated were selected over those without them. Further annotation was done using third party tRNA scanners like [ARAGORN](http://mbio-serv2.mbioekol.lu.se/ARAGORN/) and [tRNAscan](http://lowelab.ucsc.edu/tRNAscan-SE/). However, no further results were substantiated through their use.

Four genes specifically needed to be manually annotated: rps12, petB, petD, and rpl16. Rps12 is a transpliced gene, while the other three had such short inital exons that GeSeq could not annotate them. Not all mRNAs were successfully annotated by GeSeq, and were consequently manually annotated. In the final annotation, it is demonstrated that all 114 genes found in the other *Picea* chloroplasts are consered, including the 74 coding regions (CDS), 4 rRNAs, 36 tRNAs, and 15 introns (9 in CDS, 6 in tRNAs).

## MUMmer and minidot
[MUMmer](http://mummer.sourceforge.net) and [minidot](https://github.com/thackl/minidot) were used to find inverted repeats that ultimately did not make it into the final annotation. 
* See [Makefile](https://github.com/bcgsc/picea-engelmannii-plastid/blob/master/Makefile) for more information

## BLASTn
[BLASTn](https://blast.ncbi.nlm.nih.gov/Blast.cgi?PAGE_TYPE=BlastSearch) was used to align Se404-851cp genes to PG29 genes. Every subpar alignment was then analyzed in [IGV](http://software.broadinstitute.org/software/igv/) along with whole chloroplast genome alignments as well as read to assembly alignments. Most discrepancies between PG29cp and Se404-851cp were supported by the reads; those that were not were fixed as indicated by the consensus.

## table2asn_GFF
This final GFF annotation was validated using [table2asn_GFF](https://www.ncbi.nlm.nih.gov/genbank/genomes_gff):
* Generated Files: [Sequin ASN.1 file](https://github.com/bcgsc/picea-engelmannii-plastid/blob/master/table2asn_GFF/Se404-851cp.sqn), [discrepancy report](https://github.com/bcgsc/picea-engelmannii-plastid/blob/master/table2asn_GFF/Se404-851cp.dr), [error list](https://github.com/bcgsc/picea-engelmannii-plastid/blob/master/table2asn_GFF/Se404-851cp.val), [Genbank file](https://github.com/bcgsc/picea-engelmannii-plastid/blob/master/table2asn_GFF/Se404-851cp.gbf) 
* See [Makefile](https://github.com/bcgsc/picea-engelmannii-plastid/blob/master/Makefile)

## OGDraw
The .gbf file, generated by table2asn_GFF was fed through [OGDraw](https://chlorobox.mpimp-golm.mpg.de/OGDraw.html) independently
* [OGDraw settings](https://github.com/bcgsc/picea-glauca-plastid/blob/master/ogdraw/OGDRAW_settings.png)
  * The settings used for this drawing are the exact same as those used for *Picea glauca* WS77111 [chloroplast](https://github.com/bcgsc/picea-glauca-plastid)
* OGDraw files: see [ogdraw](https://github.com/bcgsc/picea-engelmannii-plastid/tree/master/ogdraw)

## Citation
Lin D, Coombe L, Jackman SD, Gagalova KK, Warren RL, Hammond SA, McDonald H, Kirk H, Pandoh P, Zhao Y, Moore RA, Mungall AJ, Ritland C, Doerksen T, Jaquish B, Bousquet J, Jones SJM, Bohlmann J, Birol I. 2019. Complete Chloroplast Genome Sequence of an Engelmann Spruce (*Picea engelmannii*, Genotype Se404-851) from Western Canada. [Microbiol Resour Announc 8:e00382-19. doi: 10.1128/MRA.00382-19.](https://mra.asm.org/content/8/24/e00382-19)
